server/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsNodes.java

```diff
@@ -76,24 +77,38 @@
         this.fs = new FsInfo.Path();
         this.plugins = new HashSet<>();
 
-        Set<InetAddress> seenAddresses = new HashSet<>(nodeResponses.size());
         List<NodeInfo> nodeInfos = new ArrayList<>(nodeResponses.size());
         List<NodeStats> nodeStats = new ArrayList<>(nodeResponses.size());
+
+        Map<InetAddress, Set<String>> inetAddressFsPaths = new HashMap<>();
         for (ClusterStatsNodeResponse nodeResponse : nodeResponses) {
             nodeInfos.add(nodeResponse.nodeInfo());
             nodeStats.add(nodeResponse.nodeStats());
             this.versions.add(nodeResponse.nodeInfo().getVersion());
             this.plugins.addAll(nodeResponse.nodeInfo().getInfo(PluginsAndModules.class).getPluginInfos());
 
-            // now do the stats that should be deduped by hardware (implemented by ip deduping)
-            TransportAddress publishAddress =
+            if (nodeResponse.nodeStats().getFs() != null){
+                // now do the stats that should be deduped by hardware (implemented by ip deduping)
+                TransportAddress publishAddress =
                     nodeResponse.nodeInfo().getInfo(TransportInfo.class).address().publishAddress();
-            final InetAddress inetAddress = publishAddress.address().getAddress();
-            if (!seenAddresses.add(inetAddress)) {
-                continue;
-            }
-            if (nodeResponse.nodeStats().getFs() != null) {
-                this.fs.add(nodeResponse.nodeStats().getFs().getTotal());
+                final InetAddress inetAddress = publishAddress.address().getAddress();
+                Iterator<FsInfo.Path> pathIterator = nodeResponse.nodeStats().getFs().iterator();
+                Set<String> fsInfoPaths;
+                if (!inetAddressFsPaths.containsKey(inetAddress)){
```

Comment:
another super minor nit
```suggestion
                if (!inetAddressFsPaths.containsKey(inetAddress)) {
```